### DISCUSSION TITLE
Support for OpenAPI "deprecated" and extension properties

### DISCUSSION THREAD

**Discussion by AnRei123** [Azure/api-management-developer-portal#2572] [Feature request]
## Background info
An e-mail notification shall be sent to API consumers when operations are newly declared to be deprecated.

To automate triggering the e-mail notifications, we want to read and compare the Boolean values of the deprecation fields for each operation in the Swagger file.  If the value of the "deprecated" field has been newly set to "true" in comparison to the previous rollout to a production environment, an e-mail notification shall be automatically generated and sent. 

Here an image with the "deprecated" field according to the specification in the OpenAPI standard:

![SwaggerFileImportAndExportDeprecatedField](https://user-images.githubusercontent.com/58981351/112958640-f9380800-9142-11eb-8e73-d98b4c60f39c.PNG)

## Bug description

The problem is that if you import an OpenAPI specification with "deprecated" tags in the APIM portal, after exporting the OpenAPI specification, the "deprecated" fields are not there anymore. They have been removed by Azure. The information is lost.

The same issue applies for customized fields starting with the prefix "x-..." according to the OpenAPI standard. They are also removed by Azure and do not show up in the Swagger file after exporting the OpenAPI specification via the APIM portal interfaces (GUI or APIs).

## Reproduction steps

1. Add a "deprecated" field in your Swagger file for an operation above the summary.
2. Import the Swagger file as an OpenAPI specification for a dedicated API version of an APIM instance via GUI or API.
3. Export the Swagger file as an OpenAPI specification for a dedicated API version of an APIM instance via GUI or API.
4. Test if the "deprecated" field is still in the exported Swagger file. 
Test result: The "deprecated" field is not part of the Swagger file anymore!

## Expected behavior

Keep the "deprecated" field as part of the OpenAPI specification also after importing and exporting for an APIM instance via GUI and API in order to allow further backend processing and scripting based on these field values.

We urgently need a fix here  for the "deprecated" field and the customized field with the prefix "x-.." for our APIM instances.  Could you provide us here a quick fix in the next 1 to 2 weeks to allow us starting our internal tests as soon as possible?

## Is your portal managed or self-hosted?

Managed

**Comment by mikebudzynski** [comment]
@AnRei123, we're not able to provide a quick fix within 1-2 weeks.

We will be implementing support for the `deprecated` property in the coming weeks and then it needs to be rolled out with the API Management service release, which can take additional 1-2 months.

Support for custom properties is also on the roadmap. Tentative ETA is by the end of the year.

**Comment by mikebudzynski** [comment]
Referencing #1148
